plugins/woocommerce/src/Blocks/Utils/BlockHooksTrait.php

```diff
@@ -100,15 +96,38 @@ protected function is_template_part_or_pattern( $context, $area ) {
 		return ( $is_pattern || $is_template_part );
 	}
 
+	/**
+	 * Given a provided context, returns whether the context refers to the target area and isn't marked as excluded.
+	 *
+	 * @param array|\WP_Post|\WP_Block_Template $context the context to check.
+	 * @param string                            $area The area to check against before inserting.
+	 * @since $VID:$
+	 * @return boolean
+	 */
+	protected function is_target_area( $context, $area ) {
+		if ( $this->is_template_part_or_pattern( $context, $area ) && ! $this->pattern_is_excluded( $context ) ) {
+			return true;
+		}
+
+		return false;
+	}
+
 	/**
 	 * Returns whether the pattern is excluded or not
 	 *
 	 * @param array|\WP_Block_Template $context Where the block is embedded.
-	 * @param array                    $pattern_exclude_list List of pattern slugs to exclude.
 	 * @since $VID:$
 	 * @return boolean
 	 */
-	protected function pattern_is_excluded( $context, $pattern_exclude_list = array() ) {
+	protected function pattern_is_excluded( $context ) {
+		/**
+		 * A list of pattern slugs to exclude from auto-insert (useful when
+		 * there are patterns that have a very specific location for the block)
+		 *
+		 * @since $VID:$
```

Comment:
I think `$VID:$` doesn't work anymore in the monorepo, so we will need to hardcode the WooCommerce version here.